src/util/usertools.c

```diff
@@ -835,3 +835,62 @@ int sss_output_fqname(TALLOC_CTX *mem_ctx,
     talloc_zfree(tmp_ctx);
     return ret;
 }
+
+static void sss_sssd_user_uid_and_gid(uid_t *uid, gid_t *gid)
+{
+    static uid_t sssd_uid;
+    static uid_t sssd_gid;
+    static bool resolved = false;
+
+    errno_t ret;
+
+    if (! resolved) {
```

Comment:
This condition will always return true, so why are you using it? Taking into account the actual code `resolved` variable shouldn't exist.

Reply:
Not really. The resolved variable is static and if we successfully resolve the SSSD_USER it is then set to true and resolved values are used next time.

I admit that I'm not sure if this "optimization" is worth, we can resolve the user on every call. Originally I thought that it will be used in more places in the code.

Reply:
Sorry, I overlooked the `static` definition.

Reply:
Do you still think that the optimization is worth?

Reply:
Well I considered it worth because it makes the rest more readable. But based on your comment I tried it the other way and this is well readable too. So lets get rid of the original approach with static variables.